python/F3DPythonBindings.cxx

```diff
@@ -14,14 +14,59 @@
 
 namespace py = pybind11;
 
+template<typename T, size_t S>
+bool load_array(const py::handle& src, bool convert, std::array<T, S>& value)
+{
+  if (!py::isinstance<py::sequence>(src))
+  {
+    return false;
+  }
+  const py::sequence l = py::reinterpret_borrow<py::sequence>(src);
+  if (l.size() != S)
+  {
+    return false;
+  }
+
+  size_t i = 0;
+  for (auto it : l)
+  {
+    value[i++] = py::cast<T>(it);
+  }
+  return true;
+}
+
+template<>
+class py::detail::type_caster<f3d::point3_t>
+{
+public:
+  bool load(handle src, bool convert) { return load_array(src, convert, value); }
+
+  static handle cast(const f3d::point3_t& src, return_value_policy, handle /* parent */)
+  {
+    return Py_BuildValue("fff", src[0], src[1], src[2]);
```

Comment:
fff ? what does this do ?

Reply:
It's the format string to make a `(float, float, float)` tuple, see https://docs.python.org/3/c-api/arg.html#building-values
Actually reading the page after looking up the link it looks like `"ddd"` could be more correct. Nice catch!